UploadDaemon_Test/UploadTaskTest.cs

```diff
@@ -222,62 +221,36 @@ public MockUploadFactory(bool returnValue)
             this.mockUpload = new MockUpload(returnValue);
         }
 
+        /// <inheritDoc />
         public IUpload CreateUpload(Config.ConfigForProcess config, IFileSystem fileSystem)
         {
             return mockUpload;
         }
     }
 
-    private class MockUpload : IUpload
-    {
-        private readonly bool returnValue;
-
-        /// <summary>
-        /// The last version that was passed to the UploadAsnyc method or null if that method was never called.
-        /// </summary>
-        public string LastUsedVersion { get; private set; } = null;
-
-        public MockUpload(bool returnValue)
-        {
-            this.returnValue = returnValue;
-        }
-
-        /// <summary>
-        /// Fakes an upload and returns the result passed to the constructor.
-        /// </summary>
-        public Task<bool> UploadAsync(string filePath, string version)
-        {
-            LastUsedVersion = version;
-            return Task.FromResult(returnValue);
-        }
-
-        public string Describe()
-        {
-            return "MockUpload";
-        }
-
-        public Task<bool> UploadLineCoverageAsync(string originalTraceFilePath, string lineCoverageReport, RevisionFileUtils.RevisionOrTimestamp revisionOrTimestamp)
-        {
-            return Task.FromResult(returnValue);
-        }
-    }
-
     private class MockLineCoverageSynthesizer : ILineCoverageSynthesizer
     {
         private readonly bool shouldProduceCoverage;
 
         public MockLineCoverageSynthesizer(bool shouldProduceCoverage = true)
         {
             this.shouldProduceCoverage = shouldProduceCoverage;
-        }
-
-        public string ConvertToLineCoverageReport(ParsedTraceFile traceFile, string symbolDirectory, GlobPatternList assemblyPatterns)
-        {
-            if (shouldProduceCoverage)
+        }
+
+        /// <inheritDoc />
+        public Dictionary<string, FileCoverage> ConvertToLineCoverage(ParsedTraceFile traceFile, string symbolDirectory, GlobPatternList assemblyPatterns)
+        {
+            if (!shouldProduceCoverage)
+            {
+                return null;
+            }
+
+            FileCoverage fileCoverage = new FileCoverage();
+            fileCoverage.CoveredLineRanges.Add((12, 33));

```

Comment:
I don't like having these test expectations separate from the test themselves. Again, I would use a `Mock` in the tests and stub them such that they return what I need for the particular test. It's just a single method to stub, so it is done with a single `mockSynthesizer.Setup(synth => synth.ConverToLinge(arg, arg)).Return(x);` statement in each test.

Reply:
but the point of that is just to have _any_ coverage at all. The tests don't care which exact lines will be returned. Thus the only input to that class is true/false = return coverage or return null.

IMO this makes the tests much easier to read than having the same mock setup code in each test, which unnecessarily bloats the test case. I much prefer the tests to be as terse as possible so I can quickly figure out what it's doing and read details on demand if necessary